contracts/lib/StateLib.sol

```diff
@@ -68,11 +68,13 @@ library StateLib {
         id1 => state2 => [index1]
          */
         mapping(uint256 => mapping(uint256 => uint256[])) stateIndexes;
+        // exisitingTypeIds is used to keep track of existing typeIds
+        mapping(bytes2 => bool) existingTypeIds;
         // This empty reserved space is put in place to allow future versions
         // of the State contract to add new SmtData struct fields without shifting down
         // storage of upgradable contracts that use this struct as a state variable
-        // (see https://docs.openzeppelin.com/upgrades-plugins/1.x/writing-upgradeable#storage-gaps)
-        uint256[48] __gap;
+        // (see https://docs.openzeppelin.com/upgrades-plugins/1.x/writing-upgradeable#storage-gaps)//
+        uint256[47] __gap;
```

Comment:
@AndriianChestnykh please confirm that is is correct. I think yes

Reply:
Yes, this is correct. One slot for the mapping placeholder.